my/coding/commits.py

```diff
@@ -168,7 +195,7 @@ def _commits(_repos: List[Path]) -> Iterator[Commit]:
 def _cached_commits_path(p: Path) -> str:
     # compute a reduced simple filepath using the absolute path of the repo
     simple_path = ''.join(filter(lambda c: c in _allowed_letters, str(p.absolute())))
-    return cache_dir() / simple_path / '_cached_commits'
+    return str(cache_dir() / 'commits' / simple_path / '_cached_commits')
```

Comment:
wonder why mypy hasn't caught missing `str` before!

Reply:
Ah, strange. I think initially I didn't have this split into a separate function, it was just a lambda in the `@cachew` decorator, and there its fine as `cachew` expects a [`PathProvider`](https://github.com/karlicoss/cachew/blob/master/src/cachew/__init__.py#L715)? but `mypy` should've definitely caught that this wasn't returning a Path...